src/plugins/core/borders.ts

```diff
@@ -621,18 +440,31 @@ export class BordersPlugin extends CorePlugin<BordersPluginState> implements Bor
     return CommandResult.Success;
   }
 
+  private ensureHasBorder(cmd: SetBorderCommand | SetZoneBordersCommand) {
+    if (!cmd.border) return CommandResult.NoChanges;
+    return CommandResult.Success;
+  }
+
+  /**
+   * Compute the borders to add to the given zone merged.
+   */
+  private addBordersToMerge(sheetId: UID, zone: Zone) {
+    const border = {
+      ...this.getZoneExternalBorders(sheetId, zone),
+      ...removeFalsyAttributes(this.getCellBorder({ sheetId, col: zone.left, row: zone.top })),
+    };
+    this.addBorder(sheetId, zone, border, true);
+  }
+
   // ---------------------------------------------------------------------------
   // Import/Export
   // ---------------------------------------------------------------------------
 
   import(data: WorkbookData) {
-    // Borders
     if (Object.keys(data.borders || {}).length) {
       for (const sheet of data.sheets) {
-        for (const [position, borderId] of iterateItemIdsPositions(sheet.id, sheet.borders)) {
-          const { sheetId, col, row } = position;
-          const border = data.borders[borderId];
-          this.setBorder(sheetId, col, row, border, false);
+        for (const [zone, borderId] of iterateItemIdsZone(sheet.borders)) {
+          this.addBorder(sheet.id, zone, data.borders[borderId]);
```

Comment:
I think the helper `iterateItemIdsZone` is useful here. It's a bazooka pour tuer une mouche.
It's easier to understand if the 2 lines are inlined.
```suggestion
        for (const zoneXc in sheet.borders) {
          const borderId = sheet.borders[zoneXc];
          this.addBorder(sheet.id, toZone(zoneXc), data.borders[borderId]);
```